packages/cli/src/executions/execution-recovery.service.ts

```diff
@@ -73,7 +73,7 @@ export class ExecutionRecoveryService {
 			unflattenData: true,
 		});
 
-		if (!execution || execution.status === 'success') return null;
+		if (!execution || (execution.status === 'success' && execution.data)) return null;
```

Comment:
what if progress saving was enabled and `execution.data` is present, but just outdated, and `execution.status` was updated correctly?

Reply:
That's an interesting case!

On master, if an execution is saving progress and finishes and updates status to `success` and then crashes before saving data, we currently do not recover it because of `execution.status === 'success'` And on this branch, in this edge case, behavior stays the same - we also won't recover it because of `execution.status === 'success' && execution.data`. 

This PR does not address this as it is about successful data-less executions. Once we save in a transaction, an execution that is saving progress will end up as either `running` with in-progress data or successful with final data, and we will recover only in the former case.

Reply:
let's address this another time 👍🏽